src/query/parser/promql/parse_test.go

```diff
@@ -178,6 +178,35 @@ func TestAggregateParses(t *testing.T) {
 	}
 }
 
+var aggregationWithTagListTests = []string{
+	// different number of tags
+	"sum(up) by (t1,)",
+	"sum(up) by (t1,t2)",
+	"sum(up) without (t1)",
+	"sum(up) without (t1, t2, t3)",
+
+	// trailing comma in tag list
+	"sum(up) by (t1,)",
+	"sum(up) without (t1, t2,)",
+
+	// alternative form
+	"sum by (t) (up)",
+	"sum by (t,) (up)",
+	"sum without (t) (up)",
+	"sum without (t,) (up)",
+}
+
+func TestAggregationWithTagListDoesNotError(t *testing.T) {
+	for _, q := range aggregationWithTagListTests {
+		t.Run(q, func(t *testing.T) {
+			p, err := Parse(q, time.Second, models.NewTagOptions(), NewParseOptions())
+			require.NoError(t, err)
+			_, _, err = p.DAG()
+			require.NoError(t, err)
```

Comment:
I was considering also testing the data in resulting DAG, though most of the values are inaccessible, stored inside `aggregation.baseOp` 

https://github.com/m3db/m3/blob/7ed094e72a4ec18cf9c660cc2c55cb7028ae189f/src/query/functions/aggregation/base.go#L76-L81

Reply:
If there's an error returned, no guarantee as to what should be in the other objects since using them without checking for error would be violating that function's postconditions